osu.Game.Rulesets.Osu/Judgements/OsuHitCircleJudgementResult.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Osu.Objects;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.Judgements
+{
+    public class OsuHitCircleJudgementResult : OsuJudgementResult
+    {
+        /// <summary>
+        /// The <see cref="HitCircle"/>.
+        /// </summary>
+        public HitCircle HitCircle => (HitCircle)HitObject;
+
+        /// <summary>
+        /// The position of the player's cursor when <see cref="HitCircle"/> was hit.
+        /// </summary>
+        public Vector2? CursorPositionAtHit;
```

Comment:
Shouldn't this be named `PositionOffset`?

Reply:
I didn't because it's not really an offset. It's a full playfield coordinate position.

Reply:
Ah right, then shouldn't `HitEvent.PositionOffset` be renamed to `HitEvent.HitPosition` or similar? I thought that property for a perfect aim would be `0f`.

Reply:
Hmm, I conceptualised that one as a general positional value - could be hit position, could be a miss position, could be offset, could be absolute position, could be null for a hit, etc.

However now I'm not 100% sure on the API here - in thinking about it a bit more, rulesets that require multiple simultaneous positions (touchscreen rulesets) couldn't really make effective use of just this one property.

Reply:
My opinion here would be to declare a specific `OsuHitEvent` that holds the positional offset value instead of adding it in the base class, other custom rulesets can declare their own hit events with their own properties usable by their own statistics.

Reply:
As I mentioned in the OP, that would require an `OsuScoreInfo`/`CatchScoreInfo`/etc to make use of, which I'm also not convinced is the best way to go. I think it'd have to be a dictionary if that's the most suitable path.

Reply:
Are you fine with the naming for now too?

Reply:
Ah, sort of glossed over that discussion. `Offset` sort of implies relative positioning for me too. Just `Position` would probably be ok for now if we're going for a general thing.